test/integration/self-provider-erc20collateralpool.ts

```diff
@@ -99,10 +106,388 @@ describe('SelfProvider - ERC20CollateralPool', () => {
+
+      tempLoans = await provider.contract.listLoansByLender(
+        BigInt(5),
+        BigInt(1),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(6)
+    })
+
+    it('success - limit = 10 with offset 0, 10, 20, ..., 90', async () => {
+      const warehouseLoans = new Array<Lend>()
+
+      let tempLoans = await provider.contract.listLoansByLender(
+        BigInt(0),
+        BigInt(10),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(10)
+
+      tempLoans = await provider.contract.listLoansByLender(
+        BigInt(10),
+        BigInt(10),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(20)
+
+      tempLoans = await provider.contract.listLoansByLender(
+        BigInt(20),
+        BigInt(10),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(30)
+
+      tempLoans = await provider.contract.listLoansByLender(
+        BigInt(30),
+        BigInt(10),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(40)
+
+      tempLoans = await provider.contract.listLoansByLender(
+        BigInt(40),
+        BigInt(10),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(50)
+
+      tempLoans = await provider.contract.listLoansByLender(
+        BigInt(50),
+        BigInt(10),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(60)
+
+      tempLoans = await provider.contract.listLoansByLender(
+        BigInt(60),
+        BigInt(10),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(70)
+
+      tempLoans = await provider.contract.listLoansByLender(
+        BigInt(70),
+        BigInt(10),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(80)
+
+      tempLoans = await provider.contract.listLoansByLender(
+        BigInt(80),
+        BigInt(10),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(90)
+
+      tempLoans = await provider.contract.listLoansByLender(
+        BigInt(90),
+        BigInt(10),
+        BigInt(0),
+        TESTING_PUBLIC_KEY
+      )
+      warehouseLoans.push(...tempLoans)
+
+      expect(warehouseLoans.length).toBe(100)
+    })
+
+    // it('success - limit = 1000', async () => {
```

Comment:
This is commented on purpose because this test requires to have 1000 loans registered but I ran out of MATIC to pay for the gas required to create the 1000 loans, so far there is around 400 loans, and as soon as I complete the register of 1000 loans, I'll un-comment this test